test/__snapshots__/test.js.snap

```diff
@@ -201,6 +210,38 @@ const iface = {
   interface: DOMImplementation,
   expose: {
     Window: { DOMImplementation }
+  },
+
+  install(globalObject) {
+    const DOMImplementation = function() {
+      throw new TypeError(\\"Illegal constructor\\");
+    };
+
+    const proto = {};
+
+    Object.defineProperties(proto, {
+      ...Object.getOwnPropertyDescriptors(iface.interface.prototype),
```

Comment:
Interesting. I'm unsure whether this will actually be faster than just doing `globalObject.DOMImplementation = { ... a bunch of generated code ... }`. And it seems like it'll have issues like `window.DOMImplementation.prototype.createDocument instanceof window.Function === false`.

But, it seems reasonable for now though as a smaller delta from what we're doing currently...

Reply:
As far as I know, creating new function every `install` invocation would have worse runtime performance characteristics than creating a brand new class constructor per `install` (like with the `[WebIDL2JSFactory]`).

I didn't want to make too many changes into a single PR, but my next goal is to get rid of the `class` and generate a property descriptor for the prototype and a property descriptor for the constructor. With this PR, the class is only used to retrieve its property descriptors. The idea would be to generate something like this in the future:

```js
const constructorDescriptors = {
    /* generated descriptors */
};
const prototypeDescriptors = {
    /* generated descriptors */
};

function install() {
    const ctor = function Node() {
        throw new TypeError('Illegal constructor');
    };
    Object.defineProperties(ctor, constructorDescriptors);

    const proto = {};
    Object.defineProperties(proto, constructorDescriptors);

    Object.defineProperty(ctor, 'prototype', { value: proto });
    Object.defineProperty(proto, 'constructor', { value: ctor });
}
```

Reply:
I'd like to not pursue any complicated attempts at sharing descriptors (and, notably, sharing the functions that are the `value` property of descriptors) without some benchmarks showing that's better than just creating a new copy each time. Engines are smart; I suspect that all these descriptor tricks could actually be less optimized than just

```js
function install() {
  class Node {
    // everything goes inside here
  }
}
```